crates/core/src/operations/write.rs

```diff
@@ -816,6 +845,34 @@ impl std::future::IntoFuture for WriteBuilder {
                         .or_else(|_| snapshot.arrow_schema())
                         .unwrap_or(schema.clone());
 
+                    let configuration = snapshot.metadata().configuration.clone();
+                    let current_protocol = snapshot.protocol();
+                    let maybe_new_protocol = if PROTOCOL.contains_timestampntz(
+                        TryInto::<StructType>::try_into(schema.clone())?.fields(),
+                    ) && !current_protocol
+                        .reader_features
+                        .clone()
```

Comment:
Do you need to clone the protocol to just check for a key in the features?

Reply:
When I didn't clone it would move the features value out of the protocol and then complain that Protocol l doesn't have the copy trait